[prefer-function-type] false positive on global augmentation

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-function-type": "error"
  }
}
```

```TS
export {}
declare global {
    interface Function {
        (...args: any[]): any
    }
}
```

**Expected Result**

No errors because the type alias with a function type causes a syntax error (`Duplicate identifier 'Function'`).

**Actual Result**

```
  5:9  error  Interface has only a call signature - use 'type Function = (...args: any[]) => any' instead  @typescript-eslint/prefer-function-type
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.7.0` |
| `@typescript-eslint/parser`        | `1.7.0` |
| `TypeScript`                       | `3.4.3` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `10.15.3` |
| `npm`                              | `6.9.0` |

--- a/packages/eslint-plugin/docs/rules/prefer-function-type.md
+++ b/packages/eslint-plugin/docs/rules/prefer-function-type.md
@@ -85,3 +85,8 @@ type Intersection = ((data: string) => number) & ((id: number) => string);
 ## When Not To Use It
 
 If you specifically want to use an interface or type literal with a single call signature for stylistic reasons, you can disable this rule.
+
+This rule has a known edge case of sometimes triggering on global augmentations such as `interface Function`.
+These edge cases are rare and often symptomatic of odd code.
+We recommend you use an [inline ESLint disable comment](https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments-1).
+See [#454](https://github.com/typescript-eslint/typescript-eslint/issues/454) for details.

